File: roguelike/roguelike_client/scripts/entities/room.cs
using System.Collections.Generic;
using Godot;

public class Room
{
    public int RommId { get; set; } = -1;
    public string RoomName { get; set; } = "";
    public Dictionary<int, Player> Players { get; set; } = new Dictionary<int, Player>();

    public Room(int roomId, string roomName)
    {
        RoomName = roomName;
        RommId = roomId;
    }

    public void AddPlayer(Player user)
    {
        Players[user.PlayerId] = user;
    }

    public void RemovePlayer(int playerId)
    {
        Players.Remove(playerId);
    }
}

File: roguelike/server_udp/src/game/room.rs
use std::{collections::HashMap, net::SocketAddr, sync::Arc};
use tokio::sync::Mutex;

use super::player::Player;

pub type RoomId = u32;
pub type RoomName = String;
pub type RoomPass = String;

#[derive(Debug)]
pub struct Room {
    pub id: RoomId,
    pub room_name: RoomName,
    pub room_pass: RoomPass,
    pub players: Mutex<HashMap<SocketAddr, Arc<Mutex<Player>>>>,
}

impl Room {
    pub fn new(
        id: RoomId,
        room_name: RoomName,
        room_pass: RoomPass,
        players: Mutex<HashMap<SocketAddr, Arc<Mutex<Player>>>>,
    ) -> Self {
        Room {
            id,
            room_name,
            room_pass,
            players,
        }
    }
}
